File: messconnect-api/src/main/java/com/messconnect/api/domain/Review.java
package com.messconnect.api.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import java.util.UUID;

/**
 * Verified review (PART 8). Created only for users with an (active/previous)
 * subscription to the mess, enforced at the API layer. Ratings are 1–5 across
 * taste, hygiene, quality and punctuality.
 */
@Entity
@Table(name = "reviews")
public class Review {

	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	private UUID id;

	@Column(name = "mess_id", nullable = false)
	private UUID messId;

	@Column(name = "user_id", nullable = false)
	private UUID userId;

	@Column(name = "subscription_id", nullable = false)
	private UUID subscriptionId;

	@Column(name = "rating_taste", nullable = false)
	private int ratingTaste;

	@Column(name = "rating_hygiene", nullable = false)
	private int ratingHygiene;

	@Column(name = "rating_quality", nullable = false)
	private int ratingQuality;

	@Column(name = "rating_punctuality", nullable = false)
	private int ratingPunctuality;

	@Column(name = "comment", length = 2000)
	private String comment;

	@Column(name = "created_at", nullable = false, updatable = false)
	private Instant createdAt = Instant.now();

	public UUID getId() {
		return id;
	}

	public void setId(UUID id) {
		this.id = id;
	}

	public UUID getMessId() {
		return messId;
	}

	public void setMessId(UUID messId) {
		this.messId = messId;
	}

	public UUID getUserId() {
		return userId;
	}

	public void setUserId(UUID userId) {
		this.userId = userId;
	}

	public UUID getSubscriptionId() {
		return subscriptionId;
	}

	public void setSubscriptionId(UUID subscriptionId) {
		this.subscriptionId = subscriptionId;
	}

	public int getRatingTaste() {
		return ratingTaste;
	}

	public void setRatingTaste(int ratingTaste) {
		this.ratingTaste = ratingTaste;
	}

	public int getRatingHygiene() {
		return ratingHygiene;
	}

	public void setRatingHygiene(int ratingHygiene) {
		this.ratingHygiene = ratingHygiene;
	}

	public int getRatingQuality() {
		return ratingQuality;
	}

	public void setRatingQuality(int ratingQuality) {
		this.ratingQuality = ratingQuality;
	}

	public int getRatingPunctuality() {
		return ratingPunctuality;
	}

	public void setRatingPunctuality(int ratingPunctuality) {
		this.ratingPunctuality = ratingPunctuality;
	}

	public String getComment() {
		return comment;
	}

	public void setComment(String comment) {
		this.comment = comment;
	}

	public Instant getCreatedAt() {
		return createdAt;
	}

	public void setCreatedAt(Instant createdAt) {
		this.createdAt = createdAt;
	}
}

File: messconnect/lib/features/reviews/models/review.dart
/// Client-side model mirroring the backend [ReviewResponse] (PART 8).
///
/// UUID/id and timestamps arrive as strings; ratings are 1-5.
class Review {
  const Review({
    required this.id,
    required this.messId,
    required this.userId,
    required this.subscriptionId,
    required this.ratingTaste,
    required this.ratingHygiene,
    required this.ratingQuality,
    required this.ratingPunctuality,
    required this.avgRating,
    required this.comment,
    required this.createdAt,
  });

  final String id;
  final String messId;
  final String userId;
  final String subscriptionId;
  final int ratingTaste;
  final int ratingHygiene;
  final int ratingQuality;
  final int ratingPunctuality;
  final double avgRating;
  final String comment;
  final DateTime createdAt;

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] as String,
      messId: json['messId'] as String,
      userId: json['userId'] as String,
      subscriptionId: json['subscriptionId'] as String,
      ratingTaste: json['ratingTaste'] as int,
      ratingHygiene: json['ratingHygiene'] as int,
      ratingQuality: json['ratingQuality'] as int,
      ratingPunctuality: json['ratingPunctuality'] as int,
      avgRating: (json['avgRating'] as num).toDouble(),
      comment: json['comment'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}